File: src/modules/ai.ts
// ─────────────────────────────────────────────────────────────
//  DSFlix SDK  –  src/modules/ai.ts
// ─────────────────────────────────────────────────────────────

import { HttpClient } from '../http';
import { AIChatResponse } from '../types';

export class AIModule {
  constructor(private http: HttpClient) {}

  /**
   * Send a message to the DSFlix AI assistant.
   * @param message - The user message to send
   * @param context - Optional system prompt or conversation context
   */
  chat(message: string, context?: string): Promise<AIChatResponse> {
    return this.http.post('/api/v2/ai/chat', { message, context });
  }
}

File: python/dsflix_sdk/modules/ai.py
from __future__ import annotations

from ..http import HttpClient


class AIModule:
    def __init__(self, http: HttpClient) -> None:
        self.http = http

    def chat(self, message: str, context: dict | None = None) -> dict:
        return self.http.post("/api/v2/ai/chat", {"message": message, "context": context})
